src/services/api-types.ts

```diff
@@ -121,7 +121,10 @@ export const GrowingVariantSchema = zod.object({
 export const LoginResponseSchema = zod.object({
   token: zod.string(),
 });
-
+export interface Selection {
```

Comment:
There is already a type for something similar in `sample-selector.ts`. You can ues `zod.infer<typeof SampleSelectorSchema>` to get it. We could name it (`type SampleSelector = ...`) in that file if you want.

For your specific usage though (showing the selected variant), I think it nicer to just pass `selectedVariantName` as `string | undefined`, since that's all the components is using anyway.

Reply:
Do you mean simply pass the name instead of the 'Selection' object?